gencli: issue with oneof with matching name

I'm not 100% sure how to explain the cause of this bug but this message:
https://github.com/googleapis/googleapis/blob/master/google/cloud/secretmanager/v1/resources.proto#L122

Doesn't produce code that compiles: https://github.com/tmc/secretsctl/runs/616003783#step:5:5

Copying the error messages here:

```
# github.com/tmc/secretsctl/cmd/secretsctl
cmd/secretsctl/update-secret.go:40:57: cannot use new(secretmanager.Replication_Automatic_) (type *secretmanager.Replication_Automatic_) as type secretmanager.Replication_Automatic in assignment
cmd/secretsctl/update-secret.go:42:59: cannot use new(secretmanager.Replication_UserManaged) (type *secretmanager.Replication_UserManaged) as type secretmanager.Replication_UserManaged in assignment
cmd/secretsctl/update-secret.go:91:54: cannot use &UpdateSecretInputSecretReplicationReplicationAutomatic (type *secretmanager.Replication_Automatic) as type secretmanager.isReplication_Replication in assignment:
	*secretmanager.Replication_Automatic does not implement secretmanager.isReplication_Replication (missing secretmanager.isReplication_Replication method)
cmd/secretsctl/update-secret.go:94:54: cannot use &UpdateSecretInputSecretReplicationReplicationUserManaged (type *secretmanager.Replication_UserManaged) as type secretmanager.isReplication_Replication in assignment:
	*secretmanager.Replication_UserManaged does not implement secretmanager.isReplication_Replication (missing secretmanager.isReplication_Replication method)
cmd/secretsctl/update-secret.go:110:60: UpdateSecretInputSecretReplicationReplicationUserManaged.UserManaged undefined (type secretmanager.Replication_UserManaged has no field or method UserManaged)
cmd/secretsctl/update-secret.go:115:11: undefined: secretmanager.Secret_LabelsEntry
cmd/secretsctl/update-secret.go:121:44: first argument to append must be slice; have map[string]string
```

Ok so the Replication_Automatic and Replication_UserManaged errors are because the oneof selector type (that implements the generated oneof interface) is named `Replication_Automatic_` and `Replication_UserManaged_` (with the trailing `_`). `go_cli` didn't account for that possibility and I believe splits on `_` to construct variable names. I believe the generated protobuf code has these trailing `_` because the containing message is `Replication`, with _nested_ messages `Automatic` and `UserManaged`, a oneof selector named `replication` and the oneof fields named `automatic` and `user_managed`. 

In the generated `go_cli`, changing the oneof selector variables like so:

```
// create-secret.go
var CreateSecretInputSecretReplicationReplicationAutomatic secretmanagerpb.Replication_Automatic_

var CreateSecretInputSecretReplicationReplicationUserManaged secretmanagerpb.Replication_UserManaged_

// update-secret.go
var UpdateSecretInputSecretReplicationReplicationAutomatic secretmanagerpb.Replication_Automatic_

var UpdateSecretInputSecretReplicationReplicationUserManaged secretmanagerpb.Replication_UserManaged_
```

resolves the related compilation errors and leaves me with the following errors (which I have yet to investigate):
```
./create-secret.go:117:11: undefined: "google.golang.org/genproto/googleapis/cloud/secretmanager/v1".Secret_LabelsEntry
./create-secret.go:123:44: first argument to append must be slice; have map[string]string
./update-secret.go:115:11: undefined: "google.golang.org/genproto/googleapis/cloud/secretmanager/v1".Secret_LabelsEntry
./update-secret.go:121:44: first argument to append must be slice; have map[string]string
```

--- a/internal/gencli/cmd_file.go
+++ b/internal/gencli/cmd_file.go
@@ -177,7 +177,7 @@ var {{$methodCmdVar}} = &cobra.Command{
 		{{ end }}
 		{{ end }}
 		{{ range .Flags }}
-		{{ if and ( .IsMessage ) .Repeated }}
+		{{ if and ( .IsMessage ) .Repeated (not .IsMap)}}
 		// unmarshal JSON strings into slice of structs
 		for _, item := range {{ .VarName }} {
 			tmp := {{ .MessageImport.Name }}.{{ .Message }}{}
@@ -187,6 +187,17 @@ var {{$methodCmdVar}} = &cobra.Command{
 			}
 
 			{{ .SliceAccessor }} = append({{ .SliceAccessor }}, &tmp)
+		}
+		{{ end }}
+		{{ if .IsMap }}
+		for _, item := range {{ .VarName }} {
+			split := strings.Split(item, "=")
+			if len(split) < 2 {
+				err = fmt.Errorf("Invalid map item: %q", item)
+				return
+			}
+
+			{{ .SliceAccessor }}[split[0]] = split[1]
 		}
 		{{ end }}
 		{{ end }}

--- a/internal/gencli/flag.go
+++ b/internal/gencli/flag.go
@@ -41,6 +41,7 @@ type Flag struct {
 	SliceAccessor string
 	IsOneOfField  bool
 	IsNested      bool
+	IsMap         bool
 	Optional      bool
 
 	// Accessor is only set after calling GenFlag

--- a/internal/gencli/gencli.go
+++ b/internal/gencli/gencli.go
@@ -461,6 +461,7 @@ func (g *gcli) buildFieldFlags(cmd *Command, msg *desc.MessageDescriptor, parent
 			Name:         prefix + field.GetName(),
 			FieldName:    title(prefix + field.GetName()),
 			Type:         field.GetType(),
+			IsMap:        field.IsMap(),
 			Repeated:     field.GetLabel() == descriptor.FieldDescriptorProto_LABEL_REPEATED,
 			IsOneOfField: isOneOf,
 			IsNested:     isInNested,
@@ -485,6 +486,13 @@ func (g *gcli) buildFieldFlags(cmd *Command, msg *desc.MessageDescriptor, parent
 
 		cmd.HasOptional = cmd.HasOptional || flag.Optional
 		cmd.HasEnums = cmd.HasEnums || flag.IsEnum()
+
+		if flag.IsMap {
+			// add "strings" import for key=value string split
+			putImport(cmd.Imports, &pbinfo.ImportSpec{
+				Path: "strings",
+			})
+		}
 
 		// build the variable name this field belongs to
 		n := title(flag.Name)

--- a/internal/gencli/gencli.go
+++ b/internal/gencli/gencli.go
@@ -492,6 +492,7 @@ func (g *gcli) buildFieldFlags(cmd *Command, msg *desc.MessageDescriptor, parent
 			putImport(cmd.Imports, &pbinfo.ImportSpec{
 				Path: "strings",
 			})
+			flag.Usage = "key=value pairs. " + flag.Usage
 		}
 
 		// build the variable name this field belongs to

